src/overlays/actors/ovl_En_Syateki_Wf/z_en_syateki_wf.c

```diff
@@ -64,60 +114,378 @@ const ActorInit En_Syateki_Wf_InitVars = {
 };
 
-#endif
-
-extern ColliderCylinderInit D_80A20E74;
-extern ColliderJntSphElementInit D_80A20E50[1];
-extern ColliderJntSphInit D_80A20EA0;
-extern ColliderCylinderInit D_80A20EB0;
-extern InitChainEntry D_80A20FBC[];
-
-extern UNK_TYPE D_0600A3CC;
+static Vec3f D_80A20FC4 = { 0.0f, 0.5f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/EnSyatekiWf_Init.s")
+static Vec3f D_80A20FD0 = { 1200.0f, 0.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/EnSyatekiWf_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A200E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A201CC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A20284.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A2030C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A20320.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A20378.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A203DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A20670.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A206DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A20710.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A2075C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A2079C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A20800.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A20858.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A208F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/EnSyatekiWf_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A20CF4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/func_80A20D10.s")
+static TexturePtr D_80A20FDC[] = {
+    object_wf_Tex_007AA8,
+    object_wf_Tex_0082A8,
+    object_wf_Tex_0084A8,
+    object_wf_Tex_0082A8,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Syateki_Wf/EnSyatekiWf_Draw.s")
+void EnSyatekiWf_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnSyatekiWf* this = THIS;
+    Path* path;
+    EnSyatekiMan* syatekiMan = (EnSyatekiMan*)this->actor.parent;
+    s32 temp;
+
+    path = syatekiMan->path;
+    if (path->unk2 != 2) {
+        do {
+            path = &globalCtx->setupPathList[path->unk1];
+        } while (path->unk2 != 2);
+    }
+
+    temp = 0;
+    if (EN_SYATEKI_WF_GET_PARAM_FF00(&this->actor) > 0) {
+        do {
+            temp++;
+            path = &globalCtx->setupPathList[path->unk1];
+        } while (temp < EN_SYATEKI_WF_GET_PARAM_FF00(&this->actor));
+    }
```

Comment:
So none of these can be for loops?

Reply:
Yeah, that seems like it ought to work, but if not, oh well.

Reply:
Yeah these do-whiles are pretty ugly, but I can't seem to find a way around them unfortunately

Reply:
Nevermind what I just said, turns out they're easy to get around; they're just while loops instead.